cloudwash/providers/ec2.py

```diff
@@ -11,65 +11,69 @@ def cleanup(**kwargs):
 
     is_dry_run = kwargs["dry_run"]
 
-    with compute_client("ec2") as ec2_client:
-        # Dry Data Collection Defs
-        def dry_vms():
-            all_vms = ec2_client.list_vms()
-            for vm in all_vms:
-                if (
-                    vm.name.startswith(settings.delete_vm)
-                    and total_running_time(vm).minutes >= settings.sla_minutes
-                ):
-                    if vm.name in settings.providers.ec2.except_vm_stop_list:
-                        dry_data["VMS"]["stop"].append(vm.name)
-                        continue
-                    dry_data["VMS"]["delete"].append(vm.name)
-            return dry_data["VMS"]
+    for region in settings.providers.ec2.region:
+        with compute_client('ec2', ec2_region=region) as ec2_client:
+            # Dry Data Collection Defs
+            def dry_vms():
+                all_vms = ec2_client.list_vms()
+                for vm in all_vms:
+                    if (
+                        vm.name.startswith(settings.delete_vm)
+                        and total_running_time(vm).minutes >= settings.sla_minutes
+                    ):
+                        if vm.name in settings.providers.ec2.except_vm_stop_list:
+                            dry_data['VMS']['stop'].append(vm.name)
+                            continue
+                        dry_data['VMS']['delete'].append(vm.name)
```

Comment:
The appending will append the `dry_data['VMS']` from multiple regions and hence while dry logging for a particular region it will show VMs from all the regions. The worst thing is when `dry_data['VMS']` undergo the deletion for every region, you will have errors saying `VM Instance Not found` becuase VMs from the previous region would have been already deleted in the loop but still appearing in the list of `dry_data['VMS']` for deletion.

Other resources will have same story for dry data display but the deletion will not be impacted because the deletion is targeted and not cumulative.